rust/datafusion/src/dataframe.rs

```diff
@@ -75,11 +75,11 @@ pub trait DataFrame: Send + Sync {
     /// # fn main() -> Result<()> {
     /// let mut ctx = ExecutionContext::new();
     /// let df = ctx.read_csv("tests/example.csv", CsvReadOptions::new())?;
-    /// let df = df.select(&[col("a") * col("b"), col("c")])?;
+    /// let df = df.select(vec![col("a") * col("b"), col("c")])?;
     /// # Ok(())
     /// # }
     /// ```
-    fn select(&self, expr: &[Expr]) -> Result<Arc<dyn DataFrame>>;
+    fn select(&self, expr: Vec<Expr>) -> Result<Arc<dyn DataFrame>>;
```

Comment:
Here is the API change -- most of the rest of this PR is updating all the call sites. Note that [`DataFrame::filter`](https://github.com/apache/arrow/pull/9692/files#diff-b31224e33fb55c5ba87aecede66d7f74d7dd425020a47804e6a8028d4809a8e6R96) takes in an owned `Expr` so taking an owned `Vec` is not that large of a departure